sys/shell/shell.c

```diff
@@ -256,6 +256,9 @@ static int readline(char *buf, size_t size)
         else {
             *line_buf_ptr++ = c;
             _putchar(c);
+#ifdef MODULE_NEWLIB
+            fflush(stdout);
+#endif
```

Comment:
> sys/shell/shell.c: otherwise the shell doesn't work correctly

I just wonder how this related to this platform/CPU.

For the doxygen and Makefile changes, I understand the purpose, but I think they shouldn't be part of this cpu-only PR, right?